beacon-chain/rpc/beaconv1/blocks_test.go

```diff
@@ -0,0 +1,399 @@
+
+	bs := &Server{
+		BeaconDB: db,
+	}
+	_, blkContainers := fillDBTestBlocks(t, ctx, db)
+	b2 := testutil.NewBeaconBlock()
+	b2.Block.Slot = 30
+	b2.Block.Body.Graffiti = bytesutil.PadTo([]byte{4}, 32)
+	require.NoError(t, db.SaveBlock(ctx, b2))
+	b3 := testutil.NewBeaconBlock()
+	b3.Block.Slot = 30
+	b3.Block.Body.Graffiti = bytesutil.PadTo([]byte{3}, 32)
+	require.NoError(t, db.SaveBlock(ctx, b3))
+
+	// Should throw an error if more than one blk returned.
+	headers, err := bs.ListBlockHeaders(ctx, &ethpb.BlockHeadersRequest{
+		Slot: 30,
+	})
+	require.NoError(t, err)
+
+	blocks := []*ethpb_alpha.SignedBeaconBlock{blkContainers[30].Block, b2, b3}
+	require.Equal(t, len(blocks), len(headers.Data))
+	for i, blk := range blocks {
+		signedHdr, err := blockutil.SignedBeaconBlockHeaderFromBlock(blk)
+		require.NoError(t, err)
+		marshaledBlkHdr, err := signedHdr.Marshal()
+		require.NoError(t, err)
+		v1alpa1BlockHdr := &ethpb_alpha.SignedBeaconBlockHeader{}
+		require.NoError(t, err)
+		require.NoError(t, proto.Unmarshal(marshaledBlkHdr, v1alpa1BlockHdr))
+		v1BlockHdr := &ethpb.SignedBeaconBlockHeader{}
+		require.NoError(t, err)
+		require.NoError(t, proto.Unmarshal(marshaledBlkHdr, v1BlockHdr))
+
+		if !reflect.DeepEqual(headers.Data[i].Header.Message, v1BlockHdr.Header) {
+			t.Error("Expected blocks to equal")
+		}
+	}
+}
+
+func TestServer_ListBlockHeaders_ParentRoot(t *testing.T) {
+	db, _ := dbTest.SetupDB(t)
+	ctx := context.Background()
+
+	bs := &Server{
+		BeaconDB: db,
+	}
+	parentRoot := bytesutil.PadTo([]byte{5, 6}, 32)
+	_, _ = fillDBTestBlocks(t, ctx, db)
+	b2 := testutil.NewBeaconBlock()
+	b2.Block.Slot = 31
+	b2.Block.ParentRoot = parentRoot
+	require.NoError(t, db.SaveBlock(ctx, b2))
+	b3 := testutil.NewBeaconBlock()
+	b3.Block.Slot = 32
+	b3.Block.ParentRoot = parentRoot
+	require.NoError(t, db.SaveBlock(ctx, b3))
+	b4 := testutil.NewBeaconBlock()
+	b4.Block.Slot = 36
+	b4.Block.ParentRoot = parentRoot
+	require.NoError(t, db.SaveBlock(ctx, b4))
+
+	// Should throw an error if more than one blk returned.
+	headers, err := bs.ListBlockHeaders(ctx, &ethpb.BlockHeadersRequest{
+		ParentRoot: parentRoot,
+	})
+	require.NoError(t, err)
+
+	blocks := []*ethpb_alpha.SignedBeaconBlock{b2, b3, b4}
+	require.Equal(t, len(blocks), len(headers.Data))
+	for i, blk := range blocks {
+		signedHdr, err := blockutil.SignedBeaconBlockHeaderFromBlock(blk)
+		require.NoError(t, err)
+		marshaledBlkHdr, err := signedHdr.Marshal()
+		require.NoError(t, err)
+		v1alpa1BlockHdr := &ethpb_alpha.SignedBeaconBlockHeader{}
+		require.NoError(t, err)
+		require.NoError(t, proto.Unmarshal(marshaledBlkHdr, v1alpa1BlockHdr))
+		v1BlockHdr := &ethpb.SignedBeaconBlockHeader{}
+		require.NoError(t, err)
+		require.NoError(t, proto.Unmarshal(marshaledBlkHdr, v1BlockHdr))
+
+		if !reflect.DeepEqual(headers.Data[i].Header.Message, v1BlockHdr.Header) {
+			t.Error("Expected blocks to equal")
+		}
+	}
+}
+
+func TestServer_ProposeBlock_OK(t *testing.T) {
```

Comment:
Can you refactor these into table driven tests? 
You have a lot of duplicated test setup and these tests are hard to understand what you are testing.

Reply:
+1